// ngx-org-chart/esm2015/lib/ngx-chart-designer/ngx-chart-designer.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, HostBinding, Input, Output } from '@angular/core';
export class NgxChartDesignerComponent {
    constructor() {
        this.hasParent = false;
        this.direction = 'vertical';
        this.itemClick = new EventEmitter();
    }
    /**
     * @return {?}
     */
    get hostClass() {
        return this.direction === 'vertical' ? 'column' : '';
    }
}
NgxChartDesignerComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-chart-designer',
                template: "<ngx-chart-node *ngIf=\"node\" ngClass=\"ngx-org-{{direction}}\" [node]=\"node\" [hasParent]=\"hasParent\"\r\n  [direction]=\"direction\" (itemClick)=\"itemClick.emit($event)\">\r\n</ngx-chart-node>\r\n\r\n<div *ngIf=\"node?.childs?.length\" ngClass=\"ngx-org-reports-{{direction}}\" class=\"ngx-org-reports\">\r\n  <ng-container *ngFor=\"let childNode of node?.childs; first as isFirst; last as isLast\">\r\n    <div ngClass=\"ngx-org-org-container-{{direction}}\" class=\"ngx-org-org-container\">\r\n      <div ngClass=\"ngx-org-connector-container-{{direction}}\" class=\"ngx-org-connector-container\">\r\n        <div class=\"ngx-org-connector ngx-org-border\" [style.border-color]=\"isFirst?'transparent':''\"></div>\r\n        <div class=\"ngx-org-border\"></div>\r\n        <div class=\"ngx-org-connector ngx-org-border\" [style.border-color]=\"isLast?'transparent':''\"></div>\r\n      </div>\r\n      <ngx-chart-designer [node]=\"childNode\" [hasParent]=\"true\" [direction]=\"direction\" (itemClick)=\"itemClick.emit($event)\">\r\n      </ngx-chart-designer>\r\n    </div>\r\n  </ng-container>\r\n</div>\r\n",
                styles: [":host{display:flex;align-items:center;flex:1}.ngx-org-vertical{flex-direction:column}.ngx-org-org-container{display:flex}.ngx-org-org-container-vertical{flex-direction:column}.ngx-org-connector{flex:1}.ngx-org-connector-container{display:flex}.ngx-org-connector-container-horizontal{flex-direction:column}.ngx-org-reports{display:flex;flex:1}.ngx-org-reports-horizontal{flex-direction:column}"]
            }] }
];
NgxChartDesignerComponent.propDecorators = {
    node: [{ type: Input }],
    hasParent: [{ type: Input }],
    direction: [{ type: Input }],
    itemClick: [{ type: Output }],
    hostClass: [{ type: HostBinding, args: ['style.flex-direction',] }]
};
if (false) {
    /** @type {?} */
    NgxChartDesignerComponent.prototype.node;
    /** @type {?} */
    NgxChartDesignerComponent.prototype.hasParent;
    /** @type {?} */
    NgxChartDesignerComponent.prototype.direction;
    /** @type {?} */
    NgxChartDesignerComponent.prototype.itemClick;
}
