// ngx-org-chart/esm5/lib/ngx-chart-node/ngx-chart-node.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
var NgxChartNodeComponent = /** @class */ (function () {
    function NgxChartNodeComponent() {
        this.hasParent = false;
        this.direction = 'vertical';
        this.itemClick = new EventEmitter();
    }
    NgxChartNodeComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-chart-node',
                    template: "<div ngClass=\"ngx-org-connector-{{direction}}\" class=\"ngx-org-border\" *ngIf=\"hasParent\"></div>\r\n<div [ngClass]=\"node?.cssClass\" class=\"ngx-org-box ngx-org-border ngx-org-background\" (click)=\"itemClick.emit(node)\">\r\n  <div *ngIf=\"node?.image\" class=\"ngx-org-image ngx-org-border\"\r\n    [ngStyle]=\"{ 'background-image': 'url(\\''+ node?.image+'\\')'}\">\r\n  </div>\r\n  <div>\r\n    <div class=\"ngx-org-name\">{{node?.name}}</div>\r\n    <div class=\"ngx-org-title\">{{node?.title}}</div>\r\n  </div>\r\n</div>\r\n<div ngClass=\"ngx-org-connector-{{direction}}\" class=\"ngx-org-border\" *ngIf=\"node?.childs?.length\"></div>\r\n",
                    styles: [":host{display:flex;align-items:center}.ngx-org-box{cursor:pointer;display:flex;align-items:center}"]
                }] }
    ];
    NgxChartNodeComponent.propDecorators = {
        node: [{ type: Input }],
        hasParent: [{ type: Input }],
        direction: [{ type: Input }],
        itemClick: [{ type: Output }]
    };
    return NgxChartNodeComponent;
}());
export { NgxChartNodeComponent };
if (false) {
    /** @type {?} */
    NgxChartNodeComponent.prototype.node;
    /** @type {?} */
    NgxChartNodeComponent.prototype.hasParent;
    /** @type {?} */
    NgxChartNodeComponent.prototype.direction;
    /** @type {?} */
    NgxChartNodeComponent.prototype.itemClick;
}
