// ngx-org-chart/esm2015/lib/ngx-org-chart/ngx-org-chart.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
export class NgxOrgChartComponent {
    constructor() {
        this.hasParent = false;
        this.direction = 'vertical';
        this.itemClick = new EventEmitter();
    }
}
NgxOrgChartComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-org-chart',
                template: "<div *ngIf=\"nodes?.length\">\r\n  <ng-container *ngFor=\"let node of nodes\">\r\n    <div class=\"ngx-org-self-{{direction}}\">\r\n      <ngx-chart-designer [node]=\"node\" [direction]=\"direction\" (itemClick)=\"itemClick.emit($event)\"></ngx-chart-designer>\r\n    </div>\r\n  </ng-container>\r\n</div>\r\n",
                styles: ["body,html{display:flex;flex:1}.ngx-org-name{font-family:\"Patua One\",cursive}.ngx-org-title{font-family:Oswald,sans-serif}.ngx-org-border{border-color:#9e9e9e}.ngx-org-box{color:#000;width:10em}.ngx-org-self-vertical{margin-bottom:2%}"]
            }] }
];
NgxOrgChartComponent.propDecorators = {
    nodes: [{ type: Input }],
    hasParent: [{ type: Input }],
    direction: [{ type: Input }],
    itemClick: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    NgxOrgChartComponent.prototype.nodes;
    /** @type {?} */
    NgxOrgChartComponent.prototype.hasParent;
    /** @type {?} */
    NgxOrgChartComponent.prototype.direction;
    /** @type {?} */
    NgxOrgChartComponent.prototype.itemClick;
}
